src/main/java/org/protege/editor/owl/server/api/ServerLayer.java

```diff
@@ -49,7 +51,13 @@ public HistoryFile createHistoryFile(String projectId, String projectName) throw
         String filename = projectName.replaceAll("\\s+","_"); // to snake-case
         return HistoryFile.createNew(rootDir, filename);
     }
-    
+
+    public void createCodegenFile(String projectId) throws IOException {
+        String rootDir = getConfiguration().getServerRoot() + File.separator + projectId;
+        String filename = rootDir + File.separator + "gencode"; // TODO: remove this constant
```

Comment:
I meant to type literal there.
Not changing the name but using a constant instead of a literal.
Just to have it in one place.